File: Projet/src/game_elements/Obstacle.java
package game_elements;

import java.util.Random;

import constants.Constants;

public class Obstacle{
    private double x, y;
    private double radius = Constants.OB_RADIUS;

    public Obstacle(){
        Random r = new Random();
        this.x = (r.nextInt(200)/100.0) - 1;
        this.y = (r.nextInt(200)/100.0) - 1;
    }

    public Obstacle(double x, double y){
        this.x = x;
        this.y = y;
    }
    

    public double getX(){
        return this.x;
    }
    
    public double getY(){
        return this.y;
    }

    public boolean isInCollisionWith(Player p){
        return  Math.sqrt(Math.pow((x - p.getX()), 2) + Math.pow(y - p.getY(), 2))  <  (this.radius + p.getRadius());
    }
    
    public boolean isInCollisionWith(double otherx, double othery, double otherradius){
        return  Math.sqrt(Math.pow((x - otherx), 2) + Math.pow(y - othery, 2))  <  (this.radius + otherradius);
    }
}

File: client/obstacle.py
import math
import pygame
from const import OB_RADIUS

class Obstacle:
    """Représente un unique obstacle dans le jeu.
    """


    def __init__(self, x, y):
        self.x = x
        self.y = y
        self.radius = OB_RADIUS
    
    def isInCollisionWith(self, player):
        distance = math.sqrt(math.pow(player.pos[0] - self.x, 2) + math.pow(player.pos[1] - self.y, 2))
        return (distance < self.radius + player.radius)

    def draw(self, window, xwindow, ywindow):
        color = (255, 255, 255)
        pygame.draw.circle(window, color, (xwindow, ywindow), int(self.radius * pygame.display.get_surface().get_width()/2))
